app/Http/Controllers/CommentsController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Comment;
+use App\Http\Requests\CommentRequest;
+use Illuminate\Http\Request;
+
+class CommentsController extends Controller
+{
+    public function __construct()
+    {
+        // You need to be logged in and have admin rights to access
+        $this->middleware('auth');
+    }
+
+    public function edit(Comment $comment, Request $request)
+    {
+        $origin = $request->query("origin", "/");
+        return view("comments.edit", compact('comment', 'origin'));
+    }
+
+    public function update(Comment $comment, CommentRequest $request)
+    {
+        if ($request->get("is_secret") && !\Auth::user()->is_admin) {
+            abort(403, 'Cannot create secret comments as non-admin');
```

Comment:
Is het niet logischer om deze check toe te voegen aan de `authorized` methode van `CommentRequest`?